### DISCUSSION TITLE
edgeID is returned as alpha numeric instead of long

### DISCUSSION THREAD

**Discussion by ChidambaramR** [JanusGraph/janusgraph#2635] [Q&A] [answered]
I use the following code to create an edge

```
        Edge e = this.g
            .V(fromId) // get vertex of id given for the source
            .as("fromVertex") // label as fromVertex to be accessed later
            .V(toId) // get  vertex of id given for destination
            .coalesce( // evaluates the provided traversals in order and returns the first traversal that emits at least one element
                inE(label) // check incoming edge of label given
                    .where( // conditional check to check if edge exists
                        outV() // get destination vertex of the edge to check
                            .as("fromVertex")), // against staged vertex
                addE(label) // add edge if not present
                    .from("fromVertex"))
            .next(); // end traversal to commit to graph

        System.out.println(Long.parseLong(e.id().toString());
```

This prints edgeId as `1lb-394-36d-38`. As a result of this, I am getting `NumberFormatException`. I thought all ids are long by default. Is there something I need to configure?

This is my current configuration

```
gremlin.graph=org.janusgraph.core.JanusGraphFactory

storage.backend=berkeleyje
storage.directory=jgex/berkeleyje

index.jgex.backend=lucene
index.jgex.directory=jgex/lucene
```

**Comment by vtslab** [comment, marked as answer]
g.E().id() returns RelationIdentifier instances, see: [https://javadoc.io/doc/org.janusgraph/janusgraph-driver/latest/org/janusgraph/graphdb/relations/RelationIdentifier.html](https://javadoc.io/doc/org.janusgraph/janusgraph-driver/latest/org/janusgraph/graphdb/relations/RelationIdentifier.html). If you want the long id, use g.E().id().getRelationId().

**Reply by ChidambaramR** [reply]
Thanks for the response. I was not aware of this

**Comment by FlorianHockmann** [comment]
>  I thought all ids are long by default.

Where did you get that? Only vertex IDs are `long`, edge ids are more complicated which is why they are represented as a `RelationIdentifier` object. Can't you just print them like this:

```java
System.out.println(e.id().toString());
```

The `RelationIdentifier` for example also contains the adjacent vertex ids.

**Reply by ChidambaramR** [reply]
Seems like it is just my assumption seeing that vertexIds are long. Hence all through my code, I typed edgeIds to long and it was failing. Thanks for clarifying.
